### PR TITLE
WT-8584 Replace wait signal with global thread counter in wt7989-compact-checkpoint-test

### PR DESCRIPTION
**Pull request by Sean04**
The wt7989-compact-checkpoint-test failed as a result of synchronizing the checkpoint and compaction threads using the conditional variable to signal when the threads are ready. The log showed the compaction thread sending a ready signal before the checkpoint thread waits for it, resulting in the checkpoint thread waiting forever. This PR replaces the signal with a global thread-ready counter as an alternative approach to force the threads to wait until all threads are ready before starting their operations.

### CODE REVIEW

File: test/csuite/wt7989_compact_checkpoint/main.c [hunk 1]
```diff
@@ -277,32 +259,26 @@ thread_func_checkpoint(void *arg)
...
-        printf("Waiting for the signal...\n");
-        /*
-         * Wait for the signal and time out after 20 seconds. wait_run_check is required because the
-         * time out is longer that one second.
-         */
-        __wt_cond_wait_signal(
-          (WT_SESSION_IMPL *)session, td->cond, 20 * WT_MILLION, wait_run_check, &signalled);
-        testutil_assert(signalled);
-        printf("Signal received!\n");
+    if (!td->stress_test) {
```
Comment (AlexeyAnisimovAU) [inline test/csuite/wt7989_compact_checkpoint/main.c:271]: This is code duplication. Let's try and extract it into a separate method. We can print "Threads ready..." messages in the caller.

Comment (Sean04) [reply test/csuite/wt7989_compact_checkpoint/main.c:266]: Fixed.

File: test/csuite/wt7989_compact_checkpoint/main.c [hunk 2]
```diff
@@ -91,21 +94,21 @@ main(int argc, char *argv[])
...
+     * Now, run test where compact and checkpoint threads are synchronized using global thread
+     * counter. Row store case.
      */
     run_test_clean(false, false, opts->preserve, opts->home, "NR", opts->uri);
 
     /*
      * Next, run test with WT_TIMING_STRESS_CHECKPOINT_SLOW. Column store case.
      */
-    run_test_clean(true, true, opts->preserve, opts->home, "SC", opts->uri);
+    // run_test_clean(true, true, opts->preserve, opts->home, "SC", opts->uri);
```
Comment (AlexeyAnisimovAU) [inline test/csuite/wt7989_compact_checkpoint/main.c:105]: I think you didn't mean to comment out this test and the one below.

File: test/csuite/wt7989_compact_checkpoint/main.c [hunk 3]
```diff
@@ -325,6 +290,23 @@ thread_func_checkpoint(void *arg)
     return (NULL);
 }
 
+/*
+ * thread_wait --
+ *     Loop to constantly yield the calling thread until all threads are ready.
+ */
+static void 
+thread_wait(void){
```
Comment (AlexeyAnisimovAU) [inline test/csuite/wt7989_compact_checkpoint/main.c:298]: Invalid location of bracket. Make sure you run `s_all` script.

**Review by AlexeyAnisimovAU — COMMENTED** [review]
Looks much better now. Couple more notes.

File: test/csuite/wt7989_compact_checkpoint/main.c (hunk 2 above)
Comment (Sean04) [reply test/csuite/wt7989_compact_checkpoint/main.c]: Ah, my mistake, thanks for that.

File: test/csuite/wt7989_compact_checkpoint/main.c (hunk 3 above)
Comment (Sean04) [reply test/csuite/wt7989_compact_checkpoint/main.c]: Fixed.

File: test/csuite/wt7989_compact_checkpoint/main.c [hunk 4]
```diff
@@ -231,15 +222,17 @@ thread_func_compact(void *arg)
 {
     struct thread_data *td;
     WT_SESSION *session;
+    // uint64_t ready_counter_local;
```
Comment (AlexeyAnisimovAU) [inline test/csuite/wt7989_compact_checkpoint/main.c:225]: Please remove this commented-out variable declaration.

**Review by AlexeyAnisimovAU — COMMENTED** [review]
Good job! One little note.

File: test/csuite/wt7989_compact_checkpoint/main.c (hunk 4 above)
Comment (Sean04) [reply test/csuite/wt7989_compact_checkpoint/main.c]: Fixed.

**Review by AlexeyAnisimovAU — APPROVED** [review]
Well done! LGTM!

**Pull request merged**